packages/sdks/src/components/block/block.helpers.ts

```diff
@@ -25,7 +27,13 @@ export const getComponent = ({
 
   const ref = registeredComponents[componentName];
 
-  if (!ref) {
+  const isRestricted =
+    model &&
+    ref?.models &&
+    ref.models?.length > 0 &&
+    !ref.models.includes(model);
```

Comment:
can we move this to a `checkIsRestrictedModel` that is reused across this PR?

Reply:
nice idea, done